server/tserver/src/main/java/org/apache/accumulo/tserver/InMemoryMap.java

```diff
@@ -531,6 +533,7 @@ public synchronized long getNumEntries() {
 
   private final Set<MemoryIterator> activeIters = Collections.synchronizedSet(new HashSet<>());
 
+  @NotThreadSafe
   class MemoryDataSource implements DataSource {
```

Comment:
A supporting class for iterators which are not thread safe, and impl shows this is intended for single threaded access (e.g., no volatile, sync, etc).